.github/workflows/main.yml

```diff
@@ -0,0 +1,42 @@
+# This is a basic workflow to help you get started with Actions
+
+name: Auto-Deploy
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ gh-pages ]
+  pull_request:
+    branches: [ gh-pages ]
```

Comment:
I think we should have `master` here since we want the action to run when a push is done to the master branch. Also, If I remember correctly, push already takes care of merging pull requests.